tests/rest/test_single_topology_rest.py

```diff
@@ -143,12 +154,18 @@ def test_single_topology_rest_solvent(mol_pair, temperature_scale_interpolation_
     ligand_conf = st.combine_confs(get_romol_conf(mol_a), get_romol_conf(mol_b))
     conf = np.concatenate([host.conf, ligand_conf])
 
-    def get_nonbonded_host_guest_ixn_energy(st: SingleTopology):
+    def compute_host_guest_ixn_energy(st: SingleTopology, ligand_idxs: set[int]):
         hgs = st.combine_with_host(host.system, lamb, host_config.num_water_atoms, st.ff, host_config.omm_topology)
-        return hgs.nonbonded_ixn_group(conf, host_config.box)
+        num_atoms_host = host.system.nonbonded_all_pairs.potential.num_atoms
+        ligand_idxs_ = np.array(list(ligand_idxs), dtype=np.int32) + num_atoms_host
+        nonbonded_ixn_group = replace(
+            hgs.nonbonded_ixn_group,
+            potential=replace(hgs.nonbonded_ixn_group.potential, row_atom_idxs=ligand_idxs_),
+        )
+        return nonbonded_ixn_group(conf, host_config.box)
 
-    U = get_nonbonded_host_guest_ixn_energy(st_rest)
-    U_ref = get_nonbonded_host_guest_ixn_energy(st)
+    U = compute_host_guest_ixn_energy(st_rest, st_rest.rest_region_atom_idxs)
+    U_ref = compute_host_guest_ixn_energy(st, st_rest.rest_region_atom_idxs)
```

Comment:
is it also worth sanity checking that ligand atoms outside of the rest region are *not* scaled by energy_scale? (in case of accidental overflows)

Reply:
Good call. Added in https://github.com/proteneer/timemachine/pull/1524/commits/39f85080b9818994be43dba89c97b8f24bbd85f3